src/attributes/val.js

```diff
@@ -128,7 +129,12 @@ jQuery.extend( {
 			set: function( elem, value ) {
 				var optionSet, option,
 					options = elem.options,
-					values = jQuery.makeArray( value ),
+
+					// Strip and collapse whitespace to normalize set values
+					// https://html.spec.whatwg.org/#strip-and-collapse-whitespace
+					values = jQuery.map( jQuery.makeArray( value ), function( val ) {
+						return jQuery.trim( val ).replace( rspaces, " " );
+					} ),
```

Comment:
To summarize, `jQuery.valHooks.select.set` should be left alone (the changes here reverted), but `jQuery.valHooks.option.get` as it exists in master (and here) has two problems:
* It is erroneously trimming `value` when derived from a content attribute.
* It is misimplementing [strip and collapse whitespace](https://html.spec.whatwg.org/multipage/infrastructure.html#strip-and-collapse-whitespace) when `value` is not derived from a content attribute.